addons/point_of_sale/models/pos_session.py

```diff
@@ -2008,6 +2008,30 @@ def get_pos_ui_res_partner_by_params(self, custom_search_params):
         partners = self.env['res.partner'].search_read(**params['search_params'])
         return partners
 
+    def load_order_pricelists(self, pricelist_ids):
+        params = {'search_params': {'domain': [('id', 'in', pricelist_ids)], 'fields': ['name', 'display_name', 'discount_policy']}}
+        pricelists = self.env['product.pricelist'].search_read(**params['search_params'])
+        for pricelist in pricelists:
+            pricelist['items'] = []
+
+        pricelist_by_id = {pricelist['id']: pricelist for pricelist in pricelists}
+        pricelist_item_domain = [('pricelist_id', 'in', [p['id'] for p in pricelists])]
+        for item in self.env['product.pricelist.item'].search_read(pricelist_item_domain, self._product_pricelist_item_fields()):
+            pricelist_by_id[item['pricelist_id'][0]]['items'].append(item)
+
+        return pricelists
+
+    def load_order_fiscal_positions(self, fp_ids):
+        params = {'search_params': {'domain': [('id', 'in', fp_ids)], 'fields': []}}
+        fps = self.env['account.fiscal.position'].search_read(**params['search_params'])
+        fiscal_position_tax_ids = sum([fpos['tax_ids'] for fpos in fps], [])
+        tax_params = {'search_params': {'domain': [('id', 'in', fiscal_position_tax_ids)], 'fields': []}}
+        fiscal_position_tax = self.env['account.fiscal.position.tax'].search_read(**tax_params['search_params'])
+        fiscal_position_by_id = {fpt['id']: fpt for fpt in fiscal_position_tax}
+        for fiscal_position in fps:
+            fiscal_position['fiscal_position_taxes_by_id'] = {tax_id: fiscal_position_by_id[tax_id] for tax_id in fiscal_position['tax_ids']}
+
+        return fps
```

Comment:
Maybe to avoid some strange side effect linked to the new mechanism of loading outside order. We can skip this and force the user to only select the pos configs which has the same pricelists and fiscal positions. We can later improve this behavior in a future PR.
Here what can be broken is the fact that external pricelist can be used on a particular customer instead of the default one just because we loaded it before once